framework/src/main/java/org/checkerframework/common/value/util/Range.java

```diff
@@ -58,20 +58,22 @@
     public static final Range NOTHING = new Range();
 
     /**
-     * Convenience construtor for a any collection of Long elements.
+     * Create a Range from a collection of Longs.
      *
-     * @param collection from which min and max values will be used as from and to values in the range.
+     * @param collection from which min and max values will be used as from and to values in the
+     *     range
      */
     public Range(Collection<Long> collection) {
         this(Collections.min(collection), Collections.max(collection));
     }
 
     /**
-     * Convenience construtor for List of Integer.
+     * Create a Range from a collection of Integers.
      *
-     * @param list from which min and max values will be used as from and to values in the range.
+     * @param collection from which min and max values will be used as from and to values in the
+     *     range
      */
-    public Range(List<Integer> list) {
+    public Range(Collection<Integer> collection) {
```

Comment:
One can not declare both convenience signatures to be Collection and Collection because of generic type erasure. Type is discarded on generics so the two signatures end up the same Range(Collection<?>) and therefore the compiler does not allow them to co-exist. Therefore on for Collection<Long> and one for List<Integer>

Reply:
You are right; my mistake.  (It's too bad that both cannot be the more general type.)